nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/FlowController.java

```diff
@@ -4034,19 +4034,29 @@ public Authorizable createDataAuthorizable(final String componentId) {
                     authorizable = new DataAuthorizable(connection.getSource());
                 }
             } else {
-                // authorizable for remote group ports should be the remote process group
-                if (connectable instanceof RemoteGroupPort) {
-                    authorizable = new DataAuthorizable(((RemoteGroupPort) connectable).getRemoteProcessGroup());
-                } else {
-                    authorizable = new DataAuthorizable(connectable);
-                }
+                authorizable = new DataAuthorizable(connectable);
             }
         }
 
         return authorizable;
     }
 
     @Override
+    public Authorizable createRemoteDataAuthorizable(String remoteGroupPortId) {
+        final DataAuthorizable authorizable;
+
+        final RemoteGroupPort remoteGroupPort = getRootGroup().findRemoteGroupPort(remoteGroupPortId);
+        if (remoteGroupPort == null) {
+            throw new ResourceNotFoundException("The component that generated this event is no longer part of the data flow.");
```

Comment:
With the refactoring it is just events. However, there is still one spot that is unused that can now be removed. I'll update.